src/main/java/com/github/maxopoly/finale/command/CardinalCommand.java

```diff
@@ -0,0 +1,65 @@
+package com.github.maxopoly.finale.command;
+
+import org.bukkit.ChatColor;
+import org.bukkit.Location;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import vg.civcraft.mc.civmodcore.command.CivCommand;
+import vg.civcraft.mc.civmodcore.command.StandaloneCommand;
+
+import java.util.LinkedList;
+import java.util.List;
+
+@CivCommand(id = "cardinal")
+public class CardinalCommand extends StandaloneCommand {
+
+	@Override
+	public boolean execute(CommandSender commandSender, String[] args) {
+		Player p = (Player) commandSender;
+		Location currLocation = p.getLocation();
+
+		if(args.length == 0){
+
+			double newYaw = Math.rint(currLocation.getYaw() / 45) * 45;
+
+			p.teleport(new Location(p.getWorld(), currLocation.getX(), currLocation.getY(),
+					currLocation.getZ(), (float) newYaw, currLocation.getPitch()));
+			return true;
+		}
+
+		String direction = args[0].toUpperCase();
+		Location l = new Location(p.getWorld(), currLocation.getX(), currLocation.getY(), currLocation.getZ());
```

Comment:
One letter variables names make for unreadable code, outside of loop counter variables they should be avoided